lld/ELF/Arch/RISCV.cpp

```diff
@@ -1251,9 +1251,8 @@ mergeAttributesSection(const SmallVector<InputSectionBase *, 0> &sections) {
     }
   }
 
-  if (hasArch) {
-    if (auto result = RISCVISAInfo::postProcessAndChecking(
-            std::make_unique<RISCVISAInfo>(xlen, exts))) {
+  if (hasArch && xlen != 0) {
```

Comment:
Note, I had to add this `xlen` check since the `xlen` is 0 if we never successfully parsed an ISA string from any object. In that case we would have generated an error earlier so the linking would fail. Maybe it would be to have a flag that indicates we found an error earlier?

Reply:
It could be useful to `errorOrWarn` after `mergeArch` call.

Reply:
The errorOrWarn is already inside mergeArch